### PR TITLE
Feature/ui component table widgets merged spec fixes

### PR DESCRIPTION
**Pull request by manwithtwowatches**

### CODE REVIEW

File: spec/features/work_packages/select_query_spec.rb [hunk 1]
```diff
@@ -54,12 +54,9 @@
...
-    context 'and the work packages menu item is clicked' do
-      before { work_packages_page.click_work_packages_menu_item }
-
-      it 'clears selected queries' do
-        work_packages_page.should_not have_selected_filter(filter_name)
-      end
+    it 'should show the filter', js: true do
+      expect(work_packages_page.selected_filter(filter_name)).not_to be_nil
     end
+
```
Comment (marutosi) [inline spec/features/work_packages/select_query_spec.rb:60]: unrelated change

Comment (manwithtwowatches) [inline spec/features/work_packages/select_query_spec.rb:60]: it's a feature that i broke whilst fixing the specs so i think it is related.

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -63,7 +63,27 @@
   # If you're not using ActiveRecord, or you'd prefer not to run each of your
   # examples within a transaction, remove the following line or assign false
   # instead of true.
-  config.use_transactional_fixtures = true
```
Comment (tilt) [inline spec/spec_helper.rb:66]: do we want to deactivate this globally or only for the cabybara+JS-based integration tests?

File: spec/features/accessibility/support/toggable_fieldsets_spec.rb [hunk 3]
```diff
@@ -46,8 +40,6 @@
 
     describe 'initial state', js: true do
       include_context 'find legend with text'
-
-      it_behaves_like 'toggle state set collapsed'
```
Comment (tilt) [inline spec/features/accessibility/support/toggable_fieldsets_spec.rb:50]: I wonder if we want to restore the original specification here, has the behaviour changed in any way?

File: spec/spec_helper.rb (hunk 2 above)
Comment (manwithtwowatches) [inline spec/spec_helper.rb:66]: hmm, i thought the code that i stole from Hagen already did it just for js specs... where did i see that before?

File: spec/features/accessibility/support/toggable_fieldsets_spec.rb (hunk 3 above)
Comment (manwithtwowatches) [inline spec/features/accessibility/support/toggable_fieldsets_spec.rb:50]: what it was looking for no longer exists. i think before whether it was collapsed or expanded was dictated by what was in fieldset-toggle-state-label but now it's done through angular so there isn't really much else to test.

Comment (tilt) [inline spec/features/accessibility/support/toggable_fieldsets_spec.rb:50]: ok, but ideally the expansions state should be as before and so the specs should work, shouldn't they?

File: spec/spec_helper.rb (hunk 2 above)
Comment (tilt) [inline spec/spec_helper.rb:66]: I don't know the branch but I think he wasn't doing this in the spec helper

File: spec/features/accessibility/support/toggable_fieldsets_spec.rb [hunk 4]
```diff
@@ -31,49 +31,48 @@
...
   shared_context 'find legend with text' do
-    let(:legend_text) { find('legend a', text: fieldset_name) }
-    let(:fieldset) { legend_text.find(:xpath, '../..') }
+    let(:legend_text) { find('legend a span', text: fieldset_name) }
+    let(:fieldset) { legend_text.find(:xpath, '../../..') }
   end
 
   shared_context 'find toggle label' do
-    let(:toggle_state_label) { legend_text.find('span.fieldset-toggle-state-label', visible: false) }
+    let(:link) { legend_text.find(:xpath, '..') }
```
Comment (manwithtwowatches) [inline spec/features/accessibility/support/toggable_fieldsets_spec.rb:39]: i know this is a bit strange but i am an xpath noob and could for the life of me figure out how to do this in one line. feel free to suggest an alternative.

File: spec/spec_helper.rb (hunk 2 above)
Comment (manwithtwowatches) [inline spec/spec_helper.rb:66]: is it even possible to change an rspec config value at run time? isn't the point that we're letting database cleaner have the responsibility of doing this and given the current setup it will indeed be transactional if it is one of the js features?

Comment (tilt) [inline spec/spec_helper.rb:66]: didn't see the example.metadata check, looks fine then!

**Comment by tilt** [comment]
works fine and specs are complete again - thx!
